src/main.cpp

```diff
@@ -690,15 +691,31 @@ void setup()
 #endif
 
 #ifdef ARCH_RASPBERRY_PI
-    PiHal *RadioLibHAL = new PiHal(1);
-    if (!rIf) {
-        rIf = new SX1262Interface((LockingArduinoHal *)RadioLibHAL, 21, 16, 18, 20);
-        if (!rIf->init()) {
-            LOG_WARN("Failed to find SX1262 radio\n");
-            delete rIf;
-            rIf = NULL;
-        } else {
-            LOG_INFO("SX1262 Radio init succeeded, using SX1262 radio\n");
+    if (settingsMap[use_sx1262]) {
+        if (!rIf) {
+            PiHal *RadioLibHAL = new PiHal(1);
+            rIf = new SX1262Interface((LockingArduinoHal *)RadioLibHAL, settingsMap[sx126x_cs], settingsMap[sx126x_dio1],
+                                      settingsMap[sx126x_reset], settingsMap[sx126x_busy]);
+            if (!rIf->init()) {
+                LOG_WARN("Failed to find SX1262 radio\n");
+                delete rIf;
+                rIf = NULL;
+            } else {
+                LOG_INFO("SX1262 Radio init succeeded, using SX1262 radio\n");
+            }
+        }
+    } else if (settingsMap[use_rf95]) {
+        if (!rIf) {
+            PiHal *RadioLibHAL = new PiHal(1);
```

Comment:
Variable PiHal goes out of scope at the end of the if-block. Where is the memory released?